src/PhpSessionPersistence.php

```diff
@@ -89,6 +93,63 @@ public function __construct(bool $nonLocking = false, bool $deleteCookieOnEmptyS
         $this->cookieSameSite = ini_get('session.cookie_samesite');
     }
 
+    /**
+     * @param array $session
+     * @return self
+     */
```

Comment:
```suggestion
     * @internal
     * @return self
     */
```

Marking this as internal means we don't have to maintain it for the next major. i.e. we can then move all this to the constructor as you originally proposed.

Reply:
Method is now final and internal.